identity-issuance/src/main/java/com/android/identity/issuance/evidence/EvidenceRequestDetailedMessage.kt

```diff
@@ -0,0 +1,26 @@
+package com.android.identity.issuance.evidence
+
+import kotlinx.io.bytestring.ByteString
+
+/**
+ * Detailed rich-text message building on top of [EvidenceRequestMessage] that additionally contains
+ * a [messageTitle] and a [messageType] that is used to identify the classification of the detailed
+ * message (such as "completed", "info", "confirmation_XYZ") that allows showing tailored UI
+ * pertaining to the message.
+ *
+ * [messageType] identifier used for classifying message types (success, error, info, ..) to show
+ *              the appropriate UI pertaining to the detailed message.
+ * [messageTitle] title to show above the message
+ * [message] message formatted as markdown
+ * [assets] images that can be referenced in markdown, type (PNG, JPEG, or SVG) determined by the extension
+ * [acceptButtonText] button label to continue to the next screen
+ * [rejectButtonText] optional button label to continue without accepting
+ */
+data class EvidenceRequestDetailedMessage(
+    val messageType: String,
```

Comment:
I'm a big fan of YAGNI. This, and the associated EvidenceRequestGenericDetailedMessageScreen, aren't currently used, and we have no plans to use them (that I'm aware of). These are just adding complexity that we don't currently need.

Seems like this would be cleaner in one of two ways:
1) Don't make this generic: just make it specific to the Completion screen. This would be something like EvidenceRequestCompletionMessage and wouldn't need messageType, and we wouldn't need the EvidenceRequestGenericDetailedMessageScreen composable.
2) Make this fully generic, so all the parameters are included in this request, including the image and whatever else. Remove the EvidenceRequestCompleteScreen composable and only have a single generic implementation that's flexible enough to cover that screen and any other detailed message screen we can think of.

I'm more a fan of option 1, despite the lack of flexibility, because with option 2 we're violating YAGNI. We can always add complexity later if we need it.

If we think there's a specific upcoming use case for the generic one, then let's discuss it.

Reply:
well alright yes I agree we don't need the generic message however if we do need it we'll have to not only create a new composable UI and new `EvidenceRequest` subclass type but we'd have to go back to code that uses `EvidenceRequestCompletionMessage` and refactor it since `EvidenceRequestCompletionMessage` will be deleted. Now you're changing the integrity of already-tested and potentially shipped code and opening a can of unforeseen-worms. But in reality this will just end up being technical debt and a new "detailed message" will be added without touching any of the existing code of the "completion message".

Leaving it as is means the complexity is higher the first time pushing this PR and subsequent uses are already baked in. Adding any new detailed messages going forward will not change the integrity of the "completion screen", you're only impacting code that is needed for the new feature requiring the new detailed messages.

If the code for "detailed message" is already in this PR and ready to use in the future without any additional complexity (only handle new message types), how important is it to refactor to `EvidenceRequestCompletionMessage` and remove the generic message view?

Reply:
Yes, it's here, and yes, we could use it for future screens...

But any code that's here needs to be maintained, and code that sits without being used tends to bit rot. Also, since we aren't using it, we can't be sure it has all the parameters we'd want to customize it, or that it supports whatever layouts we'd be interested in. Already, it's clear that it isn't flexible enough to handle the screen that we do need: the completion screen.

We could add all sorts of features, but if we have no plans to use them, then all they do is increase the maintenance burden. Do we have use cases in mind that would use the generic screen?

Reply:
fair enough and good point - it sounds like technical debit is inevitable, either from code/bit rot (providing bells and whistles when all you need is a stick) or from inability to go back and update/refactor related classes to a newer modular approach (you've got a few different sticks and want to add bells and whistles). Either way rotten code or simple classes will have to be updated/refactored. 
So no matter what, which is the lesser of the 2 evils? 

Either

Reply:
Agreed that less work upfront is the best approach, which to me means we only implement the piece we need (the completion screen), and we delete EvidenceRequestGenericDetailedMessageScreen. That approach doesn't create technical debt if we never end up needing a more flexible solution.

In the enum discussion, you mentioned a rule of thumb of wanting 3 examples before it'd make sense. The same applies here. We have two examples: our simple message screen and the new completion screen. That gives us some sense of how we could make a more flexible screen to cover all existing cases without needing a separate Composable for each. If we had a third use case, then we'd have a better idea of what parameters we'd need in a more flexible screen, and maybe we could abstract all of this away (if there's enough in common between them). ...and in that case, we would remove the completion screen and the simple message screen and replace them with the abstraction. (or keep the API but put the new abstraction in under the hood, with the simple message screen and completion screen calling directly to the flexible one with appropriate parameters)